src/qml/editorwidgets/DateTime.qml

```diff
@@ -36,7 +36,42 @@ EditorWidgetBase {
 
   property bool isDateTimeType: field.isDateOrTime
   property bool fieldIsDate: LayerUtils.fieldType( field ) === 'QDate'
-  property var currentValue: isDateTimeType ? value : Qt.formatDateTime(value, config['field_format'])
+  property var currentValue: {
+    label.text = formatDateTime( value );
```

Comment:
Shouldn't the var be set to fornafDateTime, not the result of setting label.text? It'd in theory create a loop

Reply:
Actually this is the fix that made it work for me, see:

| before | after |
|---|---|
| ![Recording 2023-11-28 at 11 29 19](https://github.com/opengisch/QField/assets/2820439/8b0a665a-73c8-42df-9664-59db469d821d) | ![Recording 2023-11-28 at 11 22 24](https://github.com/opengisch/QField/assets/2820439/cd4b40db-5009-4b41-b10e-59013730e603) |

There is indede a loop though.